### PR TITLE
[New Rule] AWS RDS Snapshot Shared with Another Account

### PR DESCRIPTION
**Pull request by imays11**
## Issues

- https://github.com/elastic/ia-trade-team/issues/346
- https://github.com/elastic/detection-rules/issues/1752

## Summary
Identifies an AWS RDS DB snapshot being shared with another AWS account. DB snapshots contain a full backup of an entire DB instance including sensitive data that can be abused if shared with unauthorized accounts or made public. 
Adversaries may use snapshots to restore a DB Instance in an environment they control as a means of data exfiltration.

Screenshot 2024-06-25 at 5 56 23 PM

### CODE REVIEW

File: rules/integrations/aws/exfiltration_rds_snapshot_shared_with_another_account.toml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    "Resources: Investigation Guide",
+    "Use Case: Threat Detection",
+    "Tactic: Exfiltration",
+]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+any where event.dataset == "aws.cloudtrail"
+    and event.provider == "rds.amazonaws.com"
```
Comment (imays11) [inline rules/integrations/aws/exfiltration_rds_snapshot_shared_with_another_account.toml:84]:
```suggestion
    and event.provider == "rds.amazonaws.com"
    and event.outcome == "success"
```

File: rules/integrations/aws/exfiltration_rds_snapshot_shared_with_another_account.toml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+creation_date = "2024/06/25"
+integration = ["aws"]
+maturity = "production"
+updated_date = "2024/06/25"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies an AWS RDS DB snapshot being shared with another AWS account. DB snapshots contain a full backup of an entire DB instance including sensitive data that can be abused if shared with unauthorized accounts or made public. 
+Adversaries may use snapshots to restore a DB Instance in an environment they control as a means of data exfiltration.
```
Comment (imays11) [inline rules/integrations/aws/exfiltration_rds_snapshot_shared_with_another_account.toml:11]:
```suggestion
Identifies an AWS RDS DB snapshot being shared with another AWS account. DB snapshots contain a full backup of an entire DB instance including sensitive data that can be abused if shared with unauthorized accounts or made public. Adversaries may use snapshots to restore a DB Instance in an environment they control as a means of data exfiltration.
```

File: rules/integrations/aws/exfiltration_rds_snapshot_shared_with_another_account.toml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+integration = ["aws"]
+maturity = "production"
+updated_date = "2024/06/25"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies an AWS RDS DB snapshot being shared with another AWS account. DB snapshots contain a full backup of an entire DB instance including sensitive data that can be abused if shared with unauthorized accounts or made public. Adversaries may use snapshots to restore a DB Instance in an environment they control as a means of data exfiltration.
+
+"""
```
Comment (imays11) [inline rules/integrations/aws/exfiltration_rds_snapshot_shared_with_another_account.toml:12]:
```suggestion
Identifies an AWS RDS DB snapshot being shared with another AWS account. DB snapshots contain a full backup of an entire DB instance including sensitive data that can be abused if shared with unauthorized accounts or made public. Adversaries may use snapshots to restore a DB Instance in an environment they control as a means of data exfiltration.
"""
```

**Review by terrancedejesus — APPROVED** [review]
Very nice addition!

**Review by Aegrah — APPROVED** [review]

**Pull request merged**